code/datums/elements/attack_knockback.dm

```diff
@@ -0,0 +1,28 @@
+/**
+ * Knocks the target back one tile when melee hitting!
+ *
+ * Used in bats and maces
+ */
+/datum/element/attack_knockback
+
+/datum/element/attack_knockback/Attach(datum/target)
+	. = ..()
+	if(!isitem(target))
+		return ELEMENT_INCOMPATIBLE
+	RegisterSignal(target, COMSIG_ITEM_ATTACK, PROC_REF(on_item_attack))
```

Comment:
Should really be using afterattack signal (just make sure to check for proximity)

`COMSIG_ITEM_ATTACK` is sent before pacifism and nobludgeon checks which means this signal may be caught even if an "attack" didn't "happen'